sklearn/ensemble/_hist_gradient_boosting/tests/test_splitting.py

```diff
@@ -856,3 +856,70 @@ def test_splitting_categorical_sanity(
     left_mask = np.isin(X_binned.ravel(), expected_categories_left)
     assert_array_equal(sample_indices[left_mask], samples_left)
     assert_array_equal(sample_indices[~left_mask], samples_right)
+
+
+def test_split_interaction_constraints():
+    """Check that allowed_features are respected."""
+    rng = np.random.RandomState(919)
```

Comment:
should we parametrize this over multiple seeds if it doesn't take too long?
Alternatively, I would suggest to handcraft a scenario where a non-constrained tree would pick either feature 1 and 2, and to assert that in the test, to show that it is indeed the interaction constrain and not just pure luck that leads to a feature in {0, 3}

Reply:
Better now with 6a0205840816525fa89bfe1a6b19527c1316a7f8?